### ISSUE TITLE
Not working on samsung s7 running Android N 7.0

### ISSUE THREAD

**Issue by praneethalla93** [leolin310148/ShortcutBadger#199]

**Comment by McLoo** [comment]
Could you check https://play.google.com/store/apps/details?id=me.leolin.isbadgeworking
and post the results?

**Comment by divyanshunegi** [comment]
Yes it does not work in Samsung Galaxy s7 with Android 7.0 , I found the sollution is to provide the class name instead of the `getApplicationContext()` or even `this`

But I am not sure how to make it work when a notification comes into a service ?

 `ShortcutBadger.applyCount(YourActivity.this, badgeCount); ` `yourActivity.this` is important in Samsung devices.

I am still searching for a solution on how to make it work for a service. I tested but it does not work properly.

**Comment by divyanshunegi** [comment]
The Issue is fixed now, It was a FCM trouble in JSON, my service methods were not firing up in Samsung devices, now fixed, so this sollution works perfectly.

`ShortcutBadger.applyCount(YourActivity.this, badgeCount); ` `yourActivity.this` is important in Samsung devices.
